test/unit/uk/gov/hmrc/bindingtariffrulingfrontend/model/PagedTest.scala

```diff
@@ -26,28 +28,130 @@ class PagedTest extends BaseSpec {
+      Paged(results = Seq(), pageIndex   = 1, pageSize = 10, resultCount = 100).pageCount shouldBe 10
+      Paged(results = Seq(), pageIndex   = 1, pageSize = 1, resultCount = 100).pageCount shouldBe 100
+      Paged(results = Seq(), resultCount = 1).pageCount shouldBe 1
+      Paged(results = Seq(), resultCount = 100).pageCount shouldBe 1
+      Paged(results = Seq(), resultCount = 1000).pageCount shouldBe 10
     }
+    // scalastyle:on magic.number
 
     "calculate nonEmpty" in {
-      Paged.empty.nonEmpty    shouldBe false
-      Paged(Seq("")).nonEmpty shouldBe true
+      Paged.empty.nonEmpty                     shouldBe false
+      Paged.empty(SimplePagination()).nonEmpty shouldBe false
+      Paged(Seq("")).nonEmpty                  shouldBe true
     }
+
+    "valid results when results are provided" in {
+      val validPaged = Json
+        .obj(
+          "results"     -> Seq(""),
+          "pageIndex"   -> 1,
+          "pageSize"    -> 1,
+          "resultCount" -> 1
+        )
+        .as[Paged[String]]
+
+      validPaged shouldBe Paged(Seq(""), 1, 1, 1)
+    }
+
+    "invalid results when no results provided" in {
+      val exception = intercept[JsResultException] {
+        Json
+          .obj(
+            "pageIndex"   -> 1,
+            "pageSize"    -> 1,
+            "resultCount" -> 1
+          )
+          .as[Paged[String]]
+      }
+
+      exception.errors.size                              shouldBe 1
+      exception.errors.flatMap(_._2.flatMap(_.messages)) shouldBe List("invalid results")
+    }
+
+    "invalid json when no pageIndex provided" in {
+      val exception = intercept[JsResultException] {
+        Json
+          .obj(
+            "results"     -> Seq(""),
+            "pageSize"    -> 1,
+            "resultCount" -> 1
+          )
+          .as[Paged[String]]
+      }
+
+      exception.errors.size                              shouldBe 1
+      exception.errors.flatMap(_._2.flatMap(_.messages)) shouldBe List("invalid pageIndex")
+    }
+
+    "invalid json when no pageSize provided" in {
+      val exception = intercept[JsResultException] {
+        Json
+          .obj(
+            "results"     -> Seq(""),
+            "pageIndex"   -> 1,
+            "resultCount" -> 1
+          )
+          .as[Paged[String]]
+      }
+
+      exception.errors.size                              shouldBe 1
+      exception.errors.flatMap(_._2.flatMap(_.messages)) shouldBe List("invalid pageSize")
+    }
+
+    "invalid json when no resultCount provided" in {
+      val exception = intercept[JsResultException] {
+        Json
+          .obj(
+            "results"   -> Seq(""),
+            "pageIndex" -> 1,
+            "pageSize"  -> 1
+          )
+          .as[Paged[String]]
+      }
+
+      exception.errors.size                              shouldBe 1
+      exception.errors.flatMap(_._2.flatMap(_.messages)) shouldBe List("invalid resultCount")
```

Comment:
You can extract this line in a method with exception as argument and re-use it everywhere, you can name it extract all json errors, this applies to all lines which are the same